pkg/apiserver/rest/metrics.go

```diff
@@ -37,10 +38,21 @@ var DualWriterOutcome = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 	NativeHistogramBucketFactor: 1.1,
 }, []string{"mode", "name", "method"})
 
-func (m *dualWriterMetrics) init() {
+func (m *dualWriterMetrics) init(reg prometheus.Registerer) {
```

Comment:
Can't we use the default registerer (with `prometheus.Register()`) instead of passing one in?

Reply:
so, _I'm not sure_. What I want to avoid is to register metrics more than once. I'm not sure if the defaultRegister is smart enough to use the existing one if one already exists. For example, the query entity was the only one which already had [metrics](https://github.com/grafana/grafana/blob/init_dualwriter_metrics/pkg/registry/apis/query/register.go#L84) and it would be good to reuse the same registerer, which collects all the metrics, instead of creating new registers per entity.

Reply:
Grafana uses the default registerer for everything afaik. I don't want to create new registerers, just to be clear. 

I just wanted to point out this might be simpler if we used the default registerer (when looking at wire, the query entity ends up using the default registerer as well).

Reply:
Thanks! I was overcomplicating this and got confused with defaults and new registries in grafana. I'll close this one and do as you suggest :)

Reply:
I guess we really have to go with the big ugl PR :( https://github.com/grafana/grafana/pull/89151#issuecomment-2165380934

Reply:
I don't think that's entierly true: yes, Grafana server at runtime does use the default registerer. But for tests it uses a different one (to let tests run in parallel and not crash on double-registration). Also, (and my memory is weak here) - the CLI part was supposed to use a no-op registerer since nobody collects metrics from CLI anyway. Thus is was suggested to inject a registerer provided by wire and pass it all the way down.